### ISSUE TITLE
MemoryEditor does not allow editing

### ISSUE THREAD

**Issue by learn-more** [ocornut/imgui_club#36]
TLDR: I can't get the memory editor to 'edit' anything, and I am unsure how to debug this.

Observed issue: When selecting a byte (both in the hex or ascii columns) the byte is briefly selected, after which the selection _seems_ to get lost again?
See screen recording:

https://github.com/ocornut/imgui_club/assets/610685/31d87bdd-939e-49a8-a7b8-dc7c46be5323

This occurs in a (minimally) edited branch of Dear ImGui, details:

Version: `v1.90.3-docking`
Backend: `imgui_impl_glfw.cpp + imgui_impl_vulkan.cpp`
Compiler/OS: `Win10 22H2 + MSVC 2022 (17.7.4)`
Full config/build information:
```
Dear ImGui 1.90.3 (19030)
--------------------------------
sizeof(size_t): 4, sizeof(ImDrawIdx): 2, sizeof(ImDrawVert): 20
define: __cplusplus=199711
define: IMGUI_DISABLE_OBSOLETE_FUNCTIONS
define: IMGUI_DISABLE_OBSOLETE_KEYIO
define: _WIN32
define: _MSC_VER=1900
define: _MSVC_LANG=201402
define: IMGUI_HAS_VIEWPORT
define: IMGUI_HAS_DOCK
--------------------------------
io.BackendPlatformName: imgui_impl_glfw
io.BackendRendererName: imgui_impl_vulkan
io.ConfigFlags: 0x00000443
 NavEnableKeyboard
 NavEnableGamepad
 DockingEnable
 ViewportsEnable
io.ConfigViewportsNoDecoration
io.ConfigInputTextCursorBlink
io.ConfigWindowsResizeFromEdges
io.ConfigMemoryCompactTimer = 60.0
io.BackendFlags: 0x0000140E
 HasMouseCursors
 HasSetMousePos
 PlatformHasViewports
 RendererHasVtxOffset
 RendererHasViewports
--------------------------------
io.Fonts: 1 fonts, Flags: 0x00000000, TexSize: 512,64
io.DisplaySize: 1280.00,720.00
io.DisplayFramebufferScale: 1.00,1.00
--------------------------------
style.WindowPadding: 8.00,8.00
style.WindowBorderSize: 1.00
style.FramePadding: 4.00,3.00
style.FrameRounding: 0.00
style.FrameBorderSize: 0.00
style.ItemSpacing: 8.00,4.00
style.ItemInnerSpacing: 4.00,4.00
```

To reproduce:
* Download `imgui_memory_editor.h` from [ea49dd3](https://github.com/ocornut/imgui_club/commit/ea49dd3c6803088d50b496e3fe981501543b7cbc) and drop it next to `imgui.h`
* Enable the attached patch
* Open `Another Window`
* Click on a byte

Patch :
```patch
$ git diff
diff --git a/examples/example_glfw_vulkan/main.cpp b/examples/example_glfw_vulkan/main.cpp
index c852e5f4d..47dd787c6 100644
--- a/examples/example_glfw_vulkan/main.cpp
+++ b/examples/example_glfw_vulkan/main.cpp
@@ -16,6 +16,7 @@
 #include "imgui.h"
 #include "imgui_impl_glfw.h"
 #include "imgui_impl_vulkan.h"
+#include "imgui_memory_editor.h"
 #include <stdio.h>          // printf, fprintf
 #include <stdlib.h>         // abort
 #define GLFW_INCLUDE_NONE
@@ -539,6 +540,9 @@ int main(int, char**)
             ImGui::Text("Hello from another window!");
             if (ImGui::Button("Close Me"))
                 show_another_window = false;
+            static MemoryEditor edit;
+            static unsigned char data[100];
+            edit.DrawContents(data, sizeof data);
             ImGui::End();
         }

```

**Comment by ocornut** [comment]
Indeed there was a bug.

In "Debug Log" I could see active id is being taken and cleared:
![image](https://github.com/ocornut/imgui_club/assets/8225057/e188d49e-bdd0-4386-991b-9e418543c7ed)

With `io.ConfigDebugIsDebuggerPresent = true` I can hover the value and break on the item:
![image](https://github.com/ocornut/imgui_club/assets/8225057/f12d8cea-d6b6-420d-8c71-d1f2bb7b273c)

And notice the window itself seems to be taking ActiveID, not the `InputText()`.

Then I followed the trail of calling SetKeyboardFocusHere() -> InputText() and stepped into it, into ItemAdd()->NavProcessItem().
Recalled the implementation of `SetKeyboardFocusHere()` changed with better tabbing support in 1.89.4, and understood the issue in `NavProcessItem()`:

```
// This is the code that is relied on for `SetKeyboardFocusHere()`.
if (g.NavMoveScoringItems && (item_flags & ImGuiItemFlags_Disabled) == 0 && (window->Flags & ImGuiWindowFlags_NoNavInputs) == 0)
{
```

Memory Editor uses `ImGuiWindowFlags_NoNavInputs` (via `ImGuiWindowFlags_NoNav`)

In other word, main lib broke `SetKeyboardFocusHere()` for windows using `ImGuiWindowFlags_NoNavInputs`.
I will work on a fix.

**Comment by ocornut** [comment]
Bug was introduced on February 8 with https://github.com/ocornut/imgui/commit/3a078466a7afdb4f47964651dbb2c4b2d19e28de

**Comment by ocornut** [comment]
This is fixed in main line by https://github.com/ocornut/imgui/commit/13d91ff9188a28ab9c906a44d6fad646fbb244e7
I will make a release tonight or tomorrow with this.
Thank you for reporting!

**Comment by ocornut** [comment]
That'll teach me for not always running tests locally, as soon as I ran them I realized the typo and fixed it: https://github.com/ocornut/imgui/commit/f5be90523d6d1f2c1104be1e097de330faa5fcf5

**Comment by learn-more** [comment]
Thanks for the detailed explanation and the fast fix!
